build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/builder/SourceGenerator.java

```diff
@@ -422,27 +426,27 @@ private void replaceXMLAttribute(String location, String tag, String attr, Strin
 		if (!featureFile.exists())
 			return;
 
-		StringBuffer buffer = null;
+		StringBuilder buffer = null;
 		try {
-			buffer = Utils.readFile(featureFile);
+			buffer = new StringBuilder(Files.readString(featureFile.toPath(), Charset.defaultCharset()));
```

Comment:
It's bad to have several places to define the charset to be used, when its supposed to use same charset.

Reply:
This is a `feature.xml` file and can have any encoding, depending on what a user sets up. So we should find that out and then use it when writing the feature file below.
Querying the workspace for the file's charset requires an additional dependency to the resources-plugin. I cannot tell if this is desired. Does anybody have a better idea?

Reply:
The best way to change XML is javax.xml.transform.TransformerFactory

Reply:
otherwise just UTF_8 -  i have never seen any non-UTF_8 xml

Reply:
damn https://bugs.eclipse.org/bugs/show_bug.cgi?id=98069 - some people write xml with other encoding

Reply:
> The best way to change XML is javax.xml.transform.TransformerFactory

I'll try that.

> damn https://bugs.eclipse.org/bugs/show_bug.cgi?id=98069 - some people write xml with other encoding

Too bad, just reading/writing it using UTF-8, would have been very convenient.